cidr-filtering-cni

```diff
@@ -0,0 +1,252 @@
+        local protocol=$(echo "$ports" | jq -r ".protocol")
+
+        if [[ "$port" == "null" || "$protocol" == "null" || "$port" == "" || "$protocol" == "" ]]; then
+            exitWithError "Port and protocol must be specified."
+        else
+            ip netns exec "$CNI_CONTAINERID" $(nft_add_rule ${table_type} ${NFT_TABLE} m-${policy_id}-${direction}-ports ${match_iface} $CNI_IFNAME ${protocol,,} dport ${port} counter meta mark set meta mark "|" 0x00010000)
+        fi
+    }
+
+    # handle ip block
+    if [[ $(echo "$stdin" | jq -r ".${direction}.subnets") != "null" ]]; then
+        for_json_array "$stdin" ".${direction}.subnets" _create_subnet_rule
+    fi
+
+    # handle ports
+    if [[ $(echo "$stdin" | jq -r ".${direction}.ports") != "null" ]]; then
+        for_json_array "$stdin" ".${direction}.ports" _accept_port
+    fi
+}
+
+exitWithError() {
+    echo "{"cniVersion": "$(jq -r ".cniVersion" <<< "$stdin")","msg":"$1","code":1,"details":"$2"}" >&3
+    exit 1
+}
+
+exitWithSuccess() {
+    prev_result=$(get_object "${stdin}" ".prevResult")
+    if [[ "$prev_result" == "null" ]]; then
+        echo "{"cniVersion": "$(jq -r ".cniVersion" <<< "$stdin")"}" >&3
+    else
+        echo "$prev_result" >&3
+    fi
+    exit 0
+}
+
+create_chains_for_filtering() {
+    local policy_id="$1"
+    local table_type="$2"
+    local direction="$3"
+
+    local match_iface="iifname"
+    if [[ "$direction" == "egress" ]]; then
+        local match_iface="oifname"
+    fi
+
+    ip netns exec "$CNI_CONTAINERID" $(create_chain "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}") || exitWithError "Failed to add chain m-${policy_id}-${direction}"
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${table_type}" "${NFT_TABLE}" m-"${direction}" "${match_iface}" "$CNI_IFNAME" counter jump m-"${policy_id}"-"${direction}") || exitWithError "Failed to add rule: jump m-${policy_id}-${direction}"
+
+    ip netns exec "$CNI_CONTAINERID" $(create_chain "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}"-subnets) || exitWithError "Failed to add chain m-"${policy_id}"-"${direction}"-subnets"
+    ip netns exec "$CNI_CONTAINERID" $(create_chain "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}"-ports) || exitWithError "Failed to add chain m-"${policy_id}"-"${direction}"-ports"
+    
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}" counter jump m-"${policy_id}"-"${direction}"-subnets) || exitWithError "Failed to add rule: jump m-"${policy_id}"-"${direction}"-subnets"
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}" counter jump m-"${policy_id}"-"${direction}"-ports) || exitWithError "Failed to add rule: jump m-"${policy_id}"-"${direction}"-ports"
+
+    create_rules_for_filtering "${policy_id}" "${table_type}" "${direction}" "${match_iface}"
+}
+
+case $CNI_COMMAND in
```

Comment:
Do we want to add it as in a main function maybe ?

Unless we can't because it is a CNI
btw there are no prints to stdout beside the final answer right ? (CNI doesnt allow it)

Reply:
> Do we want to add it as in a main function maybe ?

I'm not sure what you mean. 

> btw there are no prints to stdout beside the final answer right ? (CNI doesnt allow it)

that is correct, the only the result should be printed to stdout

Reply:
> > Do we want to add it as in a main function maybe ?
> 
> I'm not sure what you mean.

```
function main() {
}

main "$@"
```

Reply:
I see, we can do that